packages/server-renderer/src/renderToString.ts

```diff
@@ -77,7 +77,7 @@ export async function resolveTeleports(context: SSRContext) {
       // note: it's OK to await sequentially here because the Promises were
       // created eagerly in parallel.
       context.teleports[key] = await unrollBuffer(
-        (await Promise.all(context.__teleportBuffers[key])) as SSRBuffer
+        await Promise.all([context.__teleportBuffers[key]])
```

Comment:
Can I have a review on this change please, from the type this change makes sense, but from the previous code it seems the `__teleportBuffers` is `Record<string, Promise<buffer>[]>` if that's the case we must update the typing instead of casting